src/main/java/de/storchp/opentracks/osmplugin/utils/PreferencesUtils.java

```diff
@@ -2,78 +2,83 @@
 
 import android.content.Context;
 import android.content.SharedPreferences;
+import android.content.res.Resources;
 import android.net.Uri;
 import android.util.Log;
 
+import androidx.annotation.StringRes;
 import androidx.preference.PreferenceManager;
 
 import java.util.HashSet;
 import java.util.Set;
 
-import de.storchp.opentracks.osmplugin.MapsActivity;
 import de.storchp.opentracks.osmplugin.R;
 
 public class PreferencesUtils {
 
     private final static String TAG = PreferencesUtils.class.getSimpleName();
+    private static SharedPreferences sharedPrefs;
```

Comment:
Not sure if this is a good idea.
Do you any references for this approach?

Reply:
Oh it's a fantastic approach! I've tested this thoroughly with my app and other apps to see how it will work, and it's always worked flawlessly :)

- https://github.com/y20k/transistor/pull/359
- https://github.com/y20k/trackbook/pull/105
- https://github.com/y20k/escapepod/pull/83
- https://github.com/TacoTheDank/Scoop/blob/master/app/src/main/java/taco/scoop/util/PreferenceHelper.kt

Recently I found out that I can put keys in an XML file and call them using a Resource object, so now I've started doing that too. I've been searching for a solution to stop referencing strings in a hard-coded manner for at least a couple months, and all other attempts (like String.valueOf) would crash the app. However, Resources works flawlessly!

- My WIP settings rewrite of a reddit app's settings to use AndroidX Preference: https://github.com/ccrama/Slide/commits/settings-rewrite (the [latest commit](https://github.com/ccrama/Slide/commit/428c1b32cd0294cad56ab992d050af292867489b#diff-864853e48f077b513451032c7d8f0661b3cd9c470f1dc915cd791ae3023a2a93R27) shows me using Resources to call key strings)

Reply:
According to https://stackoverflow.com/a/23705667/4952072 it is save to have a static reference to the `SharedPreferences`.